### PR TITLE
Cynthia

### PR DESCRIPTION
**Pull request by cenofe**
Pull request for week 1 creation of web server. All changes are updated as of this pull request. Furthermore, implement page view servlet, however not sure if I did this correctly.

### CODE REVIEW

File: walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java [hunk 1]
```diff
@@ -26,7 +26,11 @@
 @WebServlet("/server-stats")
 public final class ServerStatsServlet extends HttpServlet {
 
-  private final Date startTime = new Date();
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
```
Comment (andiov) [inline walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java:30]: Check the indentation for both these variables and remove the /***/ above if you're not adding any comments.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -17,7 +17,7 @@
  */
 function addRandomGreeting() {
   const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+      ['I have never been on an airplane'];
```
Comment (andiov) [inline portfolio/src/main/webapp/script.js]: It's better to add more values so you can actually test if it returns different greetings.

**Review by andiov — COMMENTED** [review]
Can you add screenshots of your website to see how it's coming along? Thanks!

File: walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java (hunk 1 above)
Comment (cenofe) [reply walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java:30]: Screen Shot 2021-03-07 at 2 40 44 AM

a little gross haha but it works so yay. Also, thank you for the correction I got rid of the comments.

File: portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that responds with the current date. */
+@WebServlet("/fetch")
+public class FetchServlet extends HttpServlet {
+
+
+    private static final long serialVersionUID = 1L;
```
Comment (andiov) [inline portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java]: Check the indentation in this entire file. See how the variable name does not align with the method.

File: walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java [hunk 4]
```diff
@@ -26,9 +26,7 @@
 @WebServlet("/server-stats")
 public final class ServerStatsServlet extends HttpServlet {
 
-    /**
-     *
-     */
+
     private static final long serialVersionUID = 1L;
```
Comment (andiov) [inline walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java:30]: Check the indentation in this file as well.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -19,3 +19,35 @@ <h1>Cynthia's Website</h1>
     </div>
   </body>
 </html>
+
```
Comment (andiov) [inline portfolio/src/main/webapp/index.html:22]: Your main html file should have only HTML, one HEAD, and one BODY tag. If you have multiple scripts then you add multiple script tags. Then in your body you add all your html elements (divs, buttons, paragraphs). See example below. If you don't want to show all of them in the beginning, then you use visibility to show/hide elements. Please let me know if you have any questions.

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title></title>
    <link rel="stylesheet" href="style.css">
    <link rel="stylesheet" href="style_2.css">
    <script src="script.js"></script>
    <script src="script_1.js"></script>
    <script src="script_2.js"></script>
  </head>
  <body>
    <div id="content">
      <p>...</p>
     <button>Click me</button>
    </div>
  </body>
</html>
```

File: walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java (hunk 1 above)
Comment (andiov) [reply walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java:30]: This looks good, thanks! One thing, notice how your image is not loading. Make sure you include the image resource file and you use the correct path.

Comment (cenofe) [reply walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java:30]: okay thank you

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (cenofe) [reply portfolio/src/main/webapp/script.js]: thanks!

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -5,48 +5,38 @@
...
-<html>
-  <head>
-    <meta charset="UTF-8">
+            <title>hey</title>
+        
+            <h1>heyyyy</h1>
+            <p>Click <a href="/hey">here</a> to view content generated by a servlet.</p>
+ 
     <title>Fetch</title>
     <link rel="stylesheet" href="style.css">
```
Comment (andiov) [inline portfolio/src/main/webapp/index.html]: Stylesheets should only be in the head section, and I believe this one is already included there. The title is also set there so you should remove it from here.

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -5,48 +5,38 @@
...
-    <script src="script2.js"></script>
-  </head>
-  <body>
-    <div id="content"></div>
+  
+  
+    <div id="content">
+
```
Comment (andiov) [inline portfolio/src/main/webapp/index.html:30]: Remove empty lines that are not needed.

File: portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java [hunk 8]
```diff
@@ -15,22 +15,41 @@
...
+
 /** Servlet that responds with the current date. */
 @WebServlet("/fetch")
 public class FetchServlet extends HttpServlet {
 
 
     private static final long serialVersionUID = 1L;
 
+
     @Override
```
Comment (andiov) [inline portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java:29]: Not formatted properly.

File: portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java [hunk 9]
```diff
@@ -15,22 +15,41 @@
...
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.gson.Gson;
+
 /** Servlet that responds with the current date. */
 @WebServlet("/fetch")
 public class FetchServlet extends HttpServlet {
 
 
     private static final long serialVersionUID = 1L;
```
Comment (andiov) [inline portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java]: Not formatted properly. Please double check all files.

File: portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java [hunk 10]
```diff
@@ -15,22 +15,41 @@
...
 
+
     @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
```
Comment (andiov) [inline portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java:31]: Is this indented properly?

File: portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java (hunk 8 above)
Comment (cenofe) [reply portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java:29]: Could you elaborate on how it is supposed to be formatted?

Comment (cenofe) [reply portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java:29]: nevermind i think i fixed it in recent commits

**Comment by cenofe** [comment]
Hello, could you tell me if i am submitting pull requests correctly. Im not sure if i need to create additional branches or if this method of adding new commits to this one pull request is okay.

File: portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java (hunk 8 above)
Comment (andiov) [reply portfolio/src/main/java/com/google/sps/servlets/FetchServlet.java:29]: Hi Cynthia! Here's a good guide on style format: https://google.github.io/styleguide/javaguide.html#s4-formatting

I specifically recommend the sections on block indentation, line-wrapping, whitespace. For example in this file 
@Override is not properly aligned with the method name, and the last curly brace at the end is not indented properly. Something along the lines:
```
public class HelloWorldServlet extends HttpServlet {

  @Override
  public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
    ...
  }
}
```

For all your files (even css) make sure you remove empty lines that are not needed.

File: portfolio/src/main/java/com/google/sps/servlets/PortfolioPageViewServlet.java [hunk 11]
```diff
@@ -11,7 +11,6 @@
 @WebServlet("/hey")
 public class PortfolioPageViewServlet extends HttpServlet {
 
-    private static final long serialVersionUID = 1L;
     private int pageViews = 0;
```
Comment (andiov) [inline portfolio/src/main/java/com/google/sps/servlets/PortfolioPageViewServlet.java:14]: Same issue with indentation.

File: portfolio/src/main/webapp/style.css [hunk 12]
```diff
@@ -7,3 +7,50 @@
...
+    border: 3.4px rgb(255,0,255);
+    border-style: inset inset inset;
+    border-radius: 12px;
+     position: fixed;
+        top: 0;
+        width: 97.5%;
+        text-align: center;
+        display: inline-block;
+}
+
```
Comment (andiov) [inline portfolio/src/main/webapp/style.css:37]: Don't add any more empty lines than you need.

File: portfolio/src/main/webapp/style.css [hunk 13]
```diff
@@ -7,3 +7,50 @@
...
+  padding: 50px 16px;
+  text-decoration: none;
+}
+
+/* Links - change color on hover */
+.topnav a:hover {
+  background-color: #ddd;
+  color: black;
+}
+
```
Comment (andiov) [inline portfolio/src/main/webapp/style.css:55]: Remove empty lines.

File: portfolio/src/main/webapp/style.css [hunk 14]
```diff
@@ -7,3 +7,50 @@
 #greeting-container {
   margin-top: 20px;
 }
+
+body {
+    background: #ffffff url("https://data.whicdn.com/images/237505965/original.gif") repeat center center scroll;
+
```
Comment (andiov) [inline portfolio/src/main/webapp/style.css:13]: Remove line

**Review by andiov — COMMENTED** [review]
Regarding Pull Requests, I'm copying here what was mentioned in the chatroom:
> If you're working on multiple dependent features, you can create new branches for each feature where each additional feature is based off the previous dependent branch. This will help to avoid having multiple pull requests in the same branch with later PRs having changes from earlier PRs
> https://docs.google.com/document/d/1iSi4iCZJ_lQc7uE4S26kwOv1WPO_E-WjBvgjQTMGkv4/edit#heading=h.rg9aqwtx3xiu

For now, fix the style problems on this PR, then you can submit it. If you want to add new features, you should submit a new PR. Also, you already have one approved PR (the Readme one), please feel free to merge that in since it's already approved.

**Pull request closed without merging**